src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,138 @@ private:
             buf.writestring("?1");
             return;
         }
+        else if (sym.ident)
+        {
+            switch (sym.ident.toString)
+            {
+                case "opAssign":    buf.writestring("?4");  return;
+                case "opEquals":    buf.writestring("?8");  return;
+                case "opIndex":     buf.writestring("?A");  return;
+                case "opCall":      buf.writestring("?R");  return;
+                default:            break;
+            }
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
+            auto id = ti.tempdecl.ident;
+            const(char)* ident = id.toChars;
+
+            // test for special symbols
+            int firstTemplateArg = 0;
+            int whichOp = -1;
+            switch (ti.name.toString)
+            {
+                case "opCast":
+                    buf.writestring("?B");
+                    return;
+                case "opUnary":     whichOp = 0;    break;
+                case "opBinary":    whichOp = 1;    break;
+                case "opOpAssign":  whichOp = 2;    break;
+                case "opAssign":    whichOp = 3;    break;
+                case "opEquals":    whichOp = 4;    break;
+                case "opCall":      whichOp = 5;    break;
+                case "opIndex":     whichOp = 6;    break;
+                default:            break;
+            }
```

Comment:
Why splitting this in two switches? If we have to, I prefer an enum over an int.

Reply:
Because the block of work immediately below shouldn't be entered at all if the symbol name isn't a special name, so the names need to be considered up front.  
From there, I'd rather not repeat the string comparisons inside the block below, so I cache the comparison as an int.

I felt like defining an enum for a trivial thing was overkill, but I can do it if people feel strongly about it.

The enum definition will be about as many extra lines as the switch that will use it... I don't feel like those extra lines bring value to carry their weight.

Reply:
No, I don't feel strongly about it.